### PR TITLE
[Docs] Fixes the Contributor Guidelines

### PR DESCRIPTION
**Pull request by abhiwalia15**
## Description
Made changes to the contributor guidelines page on the major documentation here (https://openmined.github.io/PySyft/developer_guide/index.html)

## Affected Dependencies
No additional dependencies are affected or required.

## How has this been tested?
- Syntax is up-to-date
- Build an HTML page to check the content

## Checklist
- [+] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [+] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [+] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [+] My changes are covered by tests

### CODE REVIEW

File: docs/source/developer_guide/index.rst [hunk 1]
```diff
@@ -4,80 +4,154 @@
...
-All our development is done using Git and Github. If you're not too familiar with Git and Github, `start by reviewing this guide <https://guides.github.com/activities/hello-world>`_.
+Getting Started
+***************
+
+The PySyft community admires and welcomes your expertise and enthusiasm!
+We are always excited to work with new contributors and engage with them
+to build exciting new features or improve existing bugs.
+
+If you are unsure where to start or how your skills fit in, take a look
+at open ``good-first-issues`` or introduce yourself in the #introductions
```
Comment (ShubhamPalriwala) [inline docs/source/developer_guide/index.rst:19]: I think we can convert this `good-first-issues` to a link to the currently open Good First Issues on our repo. What do you say?

File: docs/source/developer_guide/index.rst [hunk 2]
```diff
@@ -4,80 +4,154 @@
...
+tracker has a lot of open-issues. Some things to keep in mind before
+picking an issue are:
 
-To contribute to any OpenMined repository you will need to `fork a repository <https://guides.github.com/activities/forking/>`_. Then you can work risk-free on your fork without affecting development on the main repository.
+-  Use labels to filter down your ideal issue.
+-  Find issues related to a bug or want some feature update.
+-  Find duplicate issues and link related ones.
 
-You will just need to fork once. After that you can call ``git fetch upstream`` and ``git pull 'branch-name'``` before you do your local changes to get the remote changes and be up-to-date.
+We recommend picking an open issue with the label ``Good first issue``
```
Comment (ShubhamPalriwala) [inline docs/source/developer_guide/index.rst:68]: Same as above

File: docs/source/developer_guide/index.rst [hunk 3]
```diff
@@ -4,80 +4,154 @@
...
+requests <https://docs.github.com/en/pull-requests>`__), so others can
+see your changes and give you feedback.
 
-To ensure code quality and make sure other people can understand your changes, you have to document your code. For documentation and general code cleanliness, we ask that you `follow the appropriate styleguide <https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md>`_ for the language you're working in.
+Before sending a PR, it is crucial to make sure you comply with the
+below instructions:
 
-Keep it DRY (Don't repeat yourself)
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+-  No commits should be made to the ``master`` branch directly.
```
Comment (ShubhamPalriwala) [inline docs/source/developer_guide/index.rst:136]: Since our default branch is "dev", prolly we can change this? What do you think?

**Review by ShubhamPalriwala — COMMENTED** [review]
Nice stuff! Just put up my suggestions for you to take a look!
Congrats on your first PR though!

File: docs/source/developer_guide/index.rst (hunk 1 above)
Comment (abhiwalia15) [reply docs/source/developer_guide/index.rst:19]: Yes, I thought the same, but currently, we have zero good first issues. We can definitely add this later when we have a list of good issues

File: docs/source/developer_guide/index.rst (hunk 3 above)
Comment (abhiwalia15) [reply docs/source/developer_guide/index.rst:136]: Makes more sense , thanks ☺️

Comment (IrinaMBejan) [reply docs/source/developer_guide/index.rst:136]: We can mention both :)

File: docs/source/developer_guide/index.rst (hunk 1 above)
Comment (IrinaMBejan) [reply docs/source/developer_guide/index.rst:19]: It's a good point, but I would say to add it now, this might fluctuate all the time anyway.

**Review by IrinaMBejan — APPROVED** [review]

**Pull request merged**